scipy/stats/_morestats.py

```diff
@@ -4791,16 +4791,18 @@ def directional_stats(samples, *, axis=0, normalize=True):
     >>> 1 - dirstats.mean_resultant_length
     0.13397459716167093
     """
-    samples = np.asarray(samples)
+    xp = array_namespace(samples)
+    samples = xp.asarray(samples)
+    
     if samples.ndim < 2:
         raise ValueError("samples must at least be two-dimensional. "
                          f"Instead samples has shape: {samples.shape!r}")
-    samples = np.moveaxis(samples, axis, 0)
+    samples = xp.moveaxis(samples, axis, 0)
     if normalize:
-        vectornorms = np.linalg.norm(samples, axis=-1, keepdims=True)
+        vectornorms = xp.linalg.vector_norm(samples, axis=-1, keepdims=True)
```

Comment:
Will need an extra guard

```
if hasattr(xp, 'linalg'):
    ...
else:
    # use `np.linalg` or `scipy.linalg`
```

Reply:
I think the most up-to-date version of this suggestion is https://github.com/scipy/scipy/pull/20794#issuecomment-2132122091, right?